File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
N = int(input())
lst = list(map(int, input().split()))
X = int(input())
lst.sort()
left = 0
right = N - 1
count = 0

while left < right:
    if lst[left] + lst[right] == X:
        left += 1
        count += 1
        right -= 1

    elif lst[left] + lst[right] < X:
        left += 1

    else:
        right -= 1

print(count)

File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
import java.io.*;
import java.util.*;

public class Main {
    static int N;
    static int[] lst;
    static int X;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        N = Integer.parseInt(br.readLine());
        lst = new int[N];
        st = new StringTokenizer(br.readLine());
        for (int i=0;i<N;i++) {
            lst[i] = Integer.parseInt(st.nextToken());
        }
        X = Integer.parseInt(br.readLine());
        Arrays.sort(lst);
        int count = 0;
        int left = 0;
        int right = N-1;
        while (left<right) {
            if (lst[left] + lst[right] == X) {
                left++;
                right--;
                count++;
            } else if (lst[left] + lst[right]<=X) {
                left++;
            } else {
                right--;
            }
        }
        System.out.println(count);
    }
}